src/main/java/de/plushnikov/intellij/plugin/processor/clazz/log/CustomLogProcessor.java

```diff
@@ -0,0 +1,155 @@
+    return ConfigDiscovery.getInstance().getStringLombokConfigProperty(ConfigKey.LOG_CUSTOM_DECLARATION, psiClass);
+  }
+
+  @NotNull
+  @Override
+  public String getLoggerType(@NotNull PsiClass psiClass) {
+    final Matcher declarationMatcher = DECLARATION_PATTERN.matcher(getCustomDeclaration(psiClass));
+    if (!declarationMatcher.matches()) {
+      return "!@#$%^&*()"; // unfortunately this is not guarded by validation; we must return something to be tested for equality
+    }
+    String loggerType = declarationMatcher.group(1);
+    if (loggerType == null) {
+      loggerType = declarationMatcher.group(2);
+    }
+    return loggerType;
+  }
+
+  @NotNull
+  @Override
+  String getLoggerInitializer(@NotNull PsiClass psiClass) {
+    final Matcher declarationMatcher = DECLARATION_PATTERN.matcher(getCustomDeclaration(psiClass));
+    if (!declarationMatcher.matches()) {
+      throw new IllegalStateException("Invalid custom log declaration.");
+    }
+    return declarationMatcher.group(2) + "." + declarationMatcher.group(3) + "(%s)";
+  }
+
+  @NotNull
+  @Override
+  List<LoggerInitializerParameter> getLoggerInitializerParameters(@NotNull PsiClass psiClass, boolean topicPresent) {
+    LoggerInitializerDeclaration declaration = parseInitializerParameters(getCustomDeclaration(psiClass));
+    if (declaration == null) {
+      throw new IllegalStateException("Invalid custom log declaration.");
+    }
+
+    if (topicPresent && declaration.withTopic == null || !topicPresent && declaration.withoutTopic == null) {
+      throw new IllegalStateException("@CustomLog is not configured to work " + (topicPresent ? "with" : "without") + " topic.");
+    }
+    return topicPresent ? declaration.withTopic : declaration.withoutTopic;
+  }
+
+  @Override
+  protected boolean validate(
+    @NotNull PsiAnnotation psiAnnotation, @NotNull PsiClass psiClass, @NotNull ProblemBuilder builder
+  ) {
+    if (!super.validate(psiAnnotation, psiClass, builder)) {
+      return false;
+    }
+
+    final LoggerInitializerDeclaration declaration = parseInitializerParameters(getCustomDeclaration(psiClass));
+    if (declaration == null) {
+      builder.addError("The @CustomLog is not configured correctly; please set log.custom.declaration in lombok.config.");
+      return false;
+    }
+    final String topic = PsiAnnotationUtil.getStringAnnotationValue(psiAnnotation, "topic");
+    final boolean topicPresent = !StringUtil.isEmptyOrSpaces(topic);
+    if (topicPresent) {
+      if (declaration.withTopic == null) {
+        builder.addError("@CustomLog does not allow a topic.");
+        return false;
+      }
+    } else {
+      if (declaration.withoutTopic == null) {
+        builder.addError("@CustomLog requires a topic.");
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Nullable
+  private LoggerInitializerDeclaration parseInitializerParameters(@NotNull String customDeclaration) {
```

Comment:
Can you move all of matcher/parse logic to one place and add some unit test for this?